File: bindings/java/src/main/java/org/indunet/robot/bus/ShutdownHandle.java
package org.indunet.robot.bus;

import com.sun.jna.Pointer;

/** Handle used to request shutdown from another thread. */
public final class ShutdownHandle implements AutoCloseable {
    private Pointer ptr;

    ShutdownHandle(Pointer ptr) {
        this.ptr = ptr;
    }

    public void shutdown() {
        RobotBusC.Holder.INSTANCE.robot_bus_shutdown_handle_shutdown(ptr);
    }

    public boolean isRunning() {
        return RobotBusC.Holder.INSTANCE.robot_bus_shutdown_handle_is_running(ptr) != 0;
    }

    @Override
    public void close() {
        RobotBusC.Holder.INSTANCE.robot_bus_shutdown_handle_free(ptr);
        ptr = null;
    }
}

File: bindings/android/src/main/kotlin/org/indunet/robot/bus/ShutdownHandle.kt
package org.indunet.robot.bus

import com.sun.jna.Pointer

/** Handle used to request shutdown from another thread. */
class ShutdownHandle internal constructor(private var ptr: Pointer?) : AutoCloseable {
    fun shutdown() {
        RobotBusC.Holder.INSTANCE.robot_bus_shutdown_handle_shutdown(ptr)
    }

    fun isRunning(): Boolean =
        RobotBusC.Holder.INSTANCE.robot_bus_shutdown_handle_is_running(ptr) != 0

    override fun close() {
        RobotBusC.Holder.INSTANCE.robot_bus_shutdown_handle_free(ptr)
        ptr = null
    }
}
